### PR TITLE
api-client get_dataset_file

### PR DESCRIPTION
**Pull request by emanuel-schmid**
Changes proposed in this PR:
- Adds a convenience method to the Client, `get_dataset_file`, which combines get_dataset_info and download_dataset, thus reducing boilerplate code.

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [x] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/util/api_client.py [hunk 1]
```diff
@@ -1071,4 +1194,27 @@ def rm_empty_dirs(directory: Path):
...
+
+    def get_dataset_file(self, **kwargs):
+        """Convenience method. Combines ``get_dataset`` and ``download_dataset``.
+        Returns the path to a single file if the dataset has only one,
+        otherwise throws an error.
+
+        Parameters
+        ----------
+        **kwargs
+            arguments for get_dataset and download_dataset
```
Comment (chahank) [inline climada/util/api_client.py:1208]:
```suggestion
            arguments for get_dataset and download_dataset
```
It is only `download_dataset` right?

**Comment by chahank** [comment]
I am a bit confused by the need of this new method. Overall, the `api_client` methods have all a bit confusing names. Could explain why this is needed?

File: climada/util/api_client.py (hunk 1 above)
Comment (emanuel-schmid) [reply climada/util/api_client.py:1208]: Yes. The ones for download_dataset are explicitly separated and all the others are submitted to get_dataset.

**Comment by emanuel-schmid** [comment]
There is no _need_ for this new method. It's nothing but a shortcut for the use case of getting a particular file from the api. So far one has to do it in two steps:

```python
ds = client.get_dataset_info(...)
_, [path] = client.download_dataset(ds, ...)
```

now it's possible in one:

```python
path = client.get_dataset_file(...)
```

**Comment by chahank** [comment]
Good for me, just need to resolve the conflict with the changelog ;).

**Comment by chahank** [comment]
good for merging.

**Pull request merged**